presto-main/src/main/java/com/facebook/presto/metadata/FunctionListBuilder.java

```diff
@@ -345,25 +361,34 @@ private static void verifyMethodSignature(Method method, TypeSignature returnTyp
             annotations = Arrays.copyOfRange(annotations, 1, annotations.length);
         }
 
+        int shift = 0;
```

Comment:
Instead of having a `shift` to offset based on `i`, I'd keep an explicit `parameterIndex` that starts at 0 and gets incremented within the loop as necessary.